### PR TITLE
feat(agent-core): add needs_planning classification to start_task

### PR DESCRIPTION
**Pull request by DanielScharfsteinGit**
https://accomplish-ai.atlassian.net/browse/ENG-155
## Summary

- When `needs_planning: false`, the agent still calls `start_task` (preserving discipline) but skips plan card, todos, and `complete_task` — it responds and stops naturally
- The completion enforcer's `isConversationalTurn()` fast-path handles natural stop (`reason: stop/end_turn`)
- Synthetic text message emitted from `complete_task.summary` so completion summaries always render in the UI
- System prompt updated with clear instructions and examples for both planning and conversational paths
- Non-task tools (`discard`, `todowrite`, `skill`, `start_task`, etc.) don't count as task work for continuation logic

### System Prompt (`config-generator.ts`)
- Introduces decision framework: `needs_planning: true` vs `false`
- JSON examples for both paths (login form, greeting, knowledge question, restaurant search)
- Explicit instruction: do NOT call `complete_task` for conversational responses

### MCP Tool (`start-task/src/index.ts`)
- `needs_planning` boolean added as required field
- `goal`, `steps`, `verification` become optional (required only when `needs_planning: true`)
- Runtime validation ensures planning tasks still provide full plan

### Adapter (`adapter.ts`)
- Marks `taskRequiresCompletion` when `needs_planning: true`
- Plan message and todos only emitted when `needs_planning: true`
- Emits synthetic text message from `complete_task.summary` so completion results always appear in chat
- No eager completion in `handleToolUse` — lets `handleProcessExit` → `completionEnforcer.handleProcessExit()` handle it (matches stable flow)
- No `hasCompleted` guard in `handleMessage` (matches stable)
- `isNonTaskContinuationTool()` excludes: `skill`, `start_task`, `discard`, `todowrite`, `complete_task`, `AskUserQuestion`, `report_checkpoint`, `report_thought`, `request_file_permission`

### Completion Enforcer (`completion-enforcer.ts`)
- `isConversationalTurn()` predicate: no task tools used + no task tools ever used + no task requires completion
- `taskToolsWereUsedEver` sticky flag prevents false conversational detection mid-task
- `markToolsUsed(countsForContinuation)` parameter for filtering non-task tools

### Tests
- Unit tests across adapter, completion-enforcer, and config-generator
- Full permutation matrix for conversational turn detection
- Integration flows for both planning and conversational paths

## Manual Test Results (v0.0.0-pr-466-20260216110311)

Tested on Accomplish desktop app with GPT 5.2, fresh clone of accomplish-enterprise repo.

### Scenario 1: Greeting — "hi"
- ✅ `needs_planning: false` — no plan card, no todos
- ✅ `step_finish action: complete` — no continuation loop
- ✅ PTY exits cleanly (code 0)

### Scenario 2: Knowledge question — "hello"
- ✅ `needs_planning: false` — no plan card, no todos
- ✅ `step_finish action: complete` — no continuation loop
- ✅ PTY exits cleanly (code 0)

### Scenario 3: File listing — "What files are in the src directory?"
- ✅ `needs_planning: true` — plan card with goal/steps/verification rendered
- ✅ Todos created and tracked through completion
- ✅ Tools used (glob, bash) — multiple `step_finish: continue` (correct)
- ✅ `complete_task(success)` called with summary
- ✅ Synthetic summary text rendered in chat UI
- ✅ Model text response also rendered after completion
- ✅ Task status shows "Completed"
- ✅ PTY exits cleanly (code 0)

### Scenario 4: Complex task — "Check my Google Calendar for tomorrow's meetings..."
- ✅ `needs_planning: true` — plan card rendered
- ✅ 13 continuation steps (expected for complex task)
- ✅ `complete_task(success)` called
- ✅ `step_finish: complete` after final text
- ✅ PTY exits cleanly (code 0)

### Cross-cutting verification
- ✅ No continuation loops on any scenario
- ✅ No adapter errors (only pre-existing Node.js FileHandle GC warnings)
- ✅ All PTY processes exit with code 0
- ✅ `needs_planning` correctly classified for all tasks
- ✅ `complete_task` only called for `needs_planning: true` tasks
- ✅ Conversational tasks complete without `complete_task`

### Bugs found and fixed during testing
1. **PTY kill after complete_task** — removed; was killing process before final text could emit
2. **`discard` tool counting as task work** — added to `isNonTaskContinuationTool()` exclusion list
3. **`hasCompleted` guard blocking messages** — removed to match stable behavior
4. **Eager completion in `handleToolUse`** — removed; stable flow through `handleProcessExit` → `completionEnforcer.handleProcessExit()` restored
5. **No completion summary in UI** — added synthetic text emission from `complete_task.summary`

## Test plan
- [x] Unit tests pass
- [x] Manual E2E: "hi" → responds without plan card or continuation loops
- [x] Manual E2E: "hello" → responds without plan card or continuation loops
- [x] Manual E2E: "What files are in the src directory?" → plan, tools, complete_task, summary in UI
- [x] Manual E2E: Complex task → plan, many continuations, complete_task, clean exit
- [x] No regressions vs stable completion flow

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a needs_planning flag to classify requests and conditionally enable planning, plan emission, and todo registration.
  * When needs_planning is false, the adapter returns a conversational response without creating a plan.

* **Behavior Changes**
  * Validation now enforces goal/steps/verification when planning is required; clearer error responses otherwise.
  * Improved continuation/completion detection and tool-counting; active background processes are cleaned up on completion.
  
* **Tests**
  * Expanded unit tests for planning logic, completion flows, and edge cases.

* **Chores**
  * Release metadata updated.

### CODE REVIEW

**Comment by orcaman** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/or.hiltch/pr-checks/a16133e2-6aee-4109-92b3-e2bebe542983?refs=9f25a4071464089d02b871e47a567d20d8cae6f6&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/or.hiltch/pr-checks/a16133e2-6aee-4109-92b3-e2bebe542983/license?refs=9f25a4071464089d02b871e47a567d20d8cae6f6&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: packages/agent-core/src/opencode/adapter.ts [hunk 1]
```diff
@@ -532,6 +532,9 @@ export class OpenCodeAdapter extends EventEmitter<OpenCodeAdapterEvents> {
     if (this.isStartTaskTool(toolName)) {
       this.startTaskCalled = true;
       const startInput = toolInput as StartTaskInput;
+      if (startInput?.needs_planning) {
+        this.completionEnforcer.markStructuredTaskStarted();
+      }
       if (startInput?.goal && startInput?.steps) {
```
[the remark this answers is not in the document]
Comment (DanielScharfsteinGit) [reply packages/agent-core/src/opencode/adapter.ts]: Fixed in ad96f803 — plan/todo emission now requires `needs_planning=true` in addition to `goal`/`steps` presence.

**Review by hairozen — APPROVED** [review]
